src/ir/proto/sql/sql_ir.proto

```diff
@@ -0,0 +1,100 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+// This is a minimal proto targeted at the MVP integration with a Google SQL
+// verifier. It's just to push a limited set of constructs across the
+// language barrier.
+syntax = "proto3";
+
+package raksha.ir.proto.sql;
+
+// This value corresponds to some source table for the query being analyzed.
+// It may have some confidentiality tags or integrity tags annotated onto it.
+// We entangle the tag claim aspect and the column identification because
+// that fits well with the policy matching procedure in the SQL verifier.
+message SourceTableColumn {
+  // The fully-qualified path to the column, including table name and (if it
+  // exists) schema name.
+  string column_path = 1;
+  repeated string confidentiality_tags = 2;
+  repeated string integrity_tags = 3;
+}
+
+// This `MergeOperation` is an all-purpose combination of inputs into a single
+// output. For the MVP, this is the only operation we will use to model our SQL
+// dataflow other than policy rule pieces.
+message MergeOperation {
+  // We may optionally associate merges with names to help with building
+  // AccessPaths. This name may include column and/or table aliases provided
+  // by the user.
+  string name = 1;
+  // Direct dataflow dependencies. Relevant for confidentiality and integrity
+  // tags.
+  repeated Value inputs = 2;
+  // Control dependencies. Confidentiality tags can flow from these inputs to
+  // the outputs. They are irrelevant to integrity tags.
+  repeated Value control_inputs = 3;
+  // Whether or not this merge actually transforms the data in some way or
+  // whether it just merges the data, keeping all possible integrity
+  // properties.
+  bool xforms_data = 4;
```

Comment:
What is the advantage of keeping the `PolicyTransform` separately? Why can't we  remove the two fields (`xforms_data` and `id`) and replace them with `PolicyTransform`? 

If you do that, you don't need any ids as well?

Reply:
Two parts to this question:

1. `xforms_data` does not mean "performs a policy transform on the data", it means "combines the data in some way that alters values or layout". You could imagine a `UNION ALL`, the list of expressions used with an `IN` operator, or a `WHERE` clause as having `xforms_data` set to `false`, while most arithmetic operations would have it set to `true`. I can change the name to remove this source of confusion.

2. We need the IDs because a `PolicyTransform` does not need to occur over only a single `MergeOperation`. Consider a pattern that attempts to match an expression comparing a timestamp against a window, like so:

```
?a - ?b > ?window
```

Let's say that the verifier is run over some SQL containing the following code:

```
col1 - col2 > 1000
```

The above expression will contain two `MergeOperation`s: one for `-` and one for `>`. The policy rule will match this expression and produce from it a single `PolicyTransform`: one which takes as inputs `col1`, `col2`, and `col3` and presumably performs some matching of `IntegrityTag`s upon them.

In this case, there is no one `MergeOperation` that has, in its scope, the result and all of the inputs to the transform. It is for this reason that we need IDs to point at different values.

(Typing out this explanation has made me realize I need a third kind of value, a literal value, and that the ID and name need to be associated with `Value`s, not just `MergeOperation`s).

Reply:
IIUC, typically there would be two merge operations: 
 - `%0 = col1 - col2`
 - `%1 =  %0 > 0`
 
However, if there is a pattern match, we will also see this as a whole during the visit in addition to the two above:
 -  `%x = col1 - col2 > 0`
 
Isn't that the case? If so, I am saying that `%0` and `%1` will effectively be dropped becausese only `%x` will be passed on to the parent of `col1 - col2 > 0`.

Reply:
About your other comment about `xforms_data`, why can't we capture `UNION_ALL` as a `Transform` also?

Reply:
That would work if there were only one policy rule. But what if there were a separate policy rule that matched:

```
?a - ?b
```

for some integrity tags? In that case, two overlapping subtrees of the original AST could be matched by separate policy rules, and destructively modifying the AST upon which the policy rules operate could cause one of the matches to be dropped.

Reply:
As we discussed on the phone, I've gotten rid of the confusing `xforms_data` field and have made the `PolicyTransform` message an `Expression` that hangs off of the normal `Expression` tree. Keeping the IDs as inputs to `PolicyTransform` to prevent mutating the tree and/or duplicating subtrees.